src/compiler/checker.ts

```diff
@@ -13277,7 +13277,7 @@ namespace ts {
                         checkAsyncFunctionReturnType(<FunctionLikeDeclaration>node);
                     }
                 }
-                if (!(<FunctionDeclaration>node).body) {
+                if (noUnusedIdentifiers && !(<FunctionDeclaration>node).body) {
                     checkUnusedTypeParameters(node);

```

Comment:
Question: Why you need to do this here? since you will register the node in this [line](https://github.com/Microsoft/TypeScript/pull/9784/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaR16132) already